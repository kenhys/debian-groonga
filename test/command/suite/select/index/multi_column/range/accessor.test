table_create Numbers TABLE_PAT_KEY Int32

table_create Pairs TABLE_HASH_KEY ShortText
column_create Pairs number1 COLUMN_SCALAR Int32
column_create Pairs number2 COLUMN_SCALAR Int32

column_create Numbers pairs \
  COLUMN_INDEX|WITH_SECTION \
  Pairs number1,number2

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos pair1 COLUMN_SCALAR Pairs
column_create Memos pair2 COLUMN_SCALAR Pairs

column_create Pairs memos \
  COLUMN_INDEX|WITH_SECTION \
  Memos pair1,pair2

load --table Pairs
[
  {
    "_key": "2-30",
    "number1": 2,
    "number2": 30
  },
  {
    "_key": "3-10",
    "number1": 3,
    "number2": 10
  },
  {
    "_key": "1-20",
    "number1": 1,
    "number2": 20
  }
]

load --table Memos
[
  {
    "_key": "300",
    "pair1": "1-20",
    "pair2": "2-30"
  },
  {
    "_key": "200",
    "pair1": "3-10",
    "pair2": "1-20"
  },
  {
    "_key": "100",
    "pair1": "2-30",
    "pair2": "3-10"
  }
]

select Memos \
  --filter 'pair2.number1 >= 2 && pair2.number2 >= 20' \
  --output_columns '_key,pair1,pair2'
