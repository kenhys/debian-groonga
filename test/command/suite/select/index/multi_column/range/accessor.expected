table_create Numbers TABLE_PAT_KEY Int32
[[0,0.0,0.0],true]
table_create Pairs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Pairs number1 COLUMN_SCALAR Int32
[[0,0.0,0.0],true]
column_create Pairs number2 COLUMN_SCALAR Int32
[[0,0.0,0.0],true]
column_create Numbers pairs   COLUMN_INDEX|WITH_SECTION   Pairs number1,number2
[[0,0.0,0.0],true]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos pair1 COLUMN_SCALAR Pairs
[[0,0.0,0.0],true]
column_create Memos pair2 COLUMN_SCALAR Pairs
[[0,0.0,0.0],true]
column_create Pairs memos   COLUMN_INDEX|WITH_SECTION   Memos pair1,pair2
[[0,0.0,0.0],true]
load --table Pairs
[
  {
    "_key": "2-30",
    "number1": 2,
    "number2": 30
  },
  {
    "_key": "3-10",
    "number1": 3,
    "number2": 10
  },
  {
    "_key": "1-20",
    "number1": 1,
    "number2": 20
  }
]
[[0,0.0,0.0],3]
load --table Memos
[
  {
    "_key": "300",
    "pair1": "1-20",
    "pair2": "2-30"
  },
  {
    "_key": "200",
    "pair1": "3-10",
    "pair2": "1-20"
  },
  {
    "_key": "100",
    "pair1": "2-30",
    "pair2": "3-10"
  }
]
[[0,0.0,0.0],3]
select Memos   --filter 'pair2.number1 >= 2 && pair2.number2 >= 20'   --output_columns '_key,pair1,pair2'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "_key",
          "ShortText"
        ],
        [
          "pair1",
          "Pairs"
        ],
        [
          "pair2",
          "Pairs"
        ]
      ],
      [
        "300",
        "1-20",
        "2-30"
      ]
    ]
  ]
]
